dotnet/src/VectorDataIntegrationTests/VectorDataIntegrationTests/Models/SimpleModel.cs

```diff
@@ -11,15 +12,34 @@ namespace VectorDataSpecificationTests.Models;
 /// <typeparam name="TKey">TKey is a generic parameter because different connectors support different key types.</typeparam>
 public sealed class SimpleModel<TKey>
 {
+    public const int DimensionCount = 3;
+
     [VectorStoreRecordKey(StoragePropertyName = "key")]
-    public TKey? Id { get; set; }
+    public TKey Id { get; set; } = default!;
 
     [VectorStoreRecordData(StoragePropertyName = "text")]
     public string? Text { get; set; }
 
     [VectorStoreRecordData(StoragePropertyName = "number")]
     public int Number { get; set; }
 
-    [VectorStoreRecordVector(Dimensions: 10, StoragePropertyName = "embedding")]
+    [VectorStoreRecordVector(Dimensions: DimensionCount, StoragePropertyName = "embedding")]
```

Comment:
We already have configuration for SimpleModdel in SimpleModelFixture for all this, no? (in GetRecordDefinition?) I think it's better to keep all configuration in over there in the fixture, since this way specific implementations can eaily override any detail, and all the config is concentrated in one place.

Reply:
Some of the tests like `CollectionConformanceTests` don't provide the record definition in explicit way (on purpose, as I expect this to be the common case for the end users).

Reply:
Hmm, OK... Yeah, we obviously need good coverage for attribute-based config - though that should be specific tests around that. Whereas other tests that don't specifically exercise attribute-based config should ideally just work with a record definition for easy management/overridability/etc.

Anyway, not very important.